src/script/cpp_api/s_base.cpp

```diff
@@ -168,6 +168,40 @@ void ScriptApiBase::loadScript(const std::string &script_path)
 	lua_pop(L, 1); // Pop error handler
 }
 
+#ifndef SERVER
+void ScriptApiBase::loadModFromMemory(const std::string &mod_name)
+{
+	ModNameStorer mod_name_storer(getStack(), mod_name);
```

Comment:
This is a ugly hack that needs to get fixed at some point (all mods need to have a separate Lua state in order for the mod name to be reliable).  But you don't have to worry about that yet.